### ISSUE TITLE
Clarification on normalization before L2 norm and student flag

### ISSUE THREAD

**Issue by Busso00** [AmirMansurian/AttnFD#5]
I read the code and the paper and I appreciate your idea, since on paper I read that normalization (of feature obtained via CBAM module,  before L2 loss) is across channel, but then I read the code and it seems that normalization is across all C, B, H, W. Am I wrong? If not which version is the one that obtained the result reported in paper.

I also wanted to ask if there is any difference between CBAM module applied on teacher feature and student, since I have seen the 'student' flag, but then it seems that you use always student version, is that correct?

Thanks in advance for your attention

**Comment by AmirMansurian** [comment]
Hi

The correct one is what we have implemented in the codes.

For your other question, first we test that CBAM be trained just for student, and in teacher side it just apply pooling operation without any learnable parameter. Then we find that training it for both side leads to better performance. So, the flag 'student' was for this reason.

Thanks for your questions. Please let me know if you still have any question.

Amir

**Comment by Busso00** [comment]
Thanks, 

I have another question:

To train it for teacher do you need another training procedure? I mean you use as target for the refined features the downscaled gt with a CE or something else? or you train both when performing kd?

Thanks again for your answer,

Federico

**Comment by AmirMansurian** [comment]
Yes, you need first train the teacher and use the pre-trained model in the training of the student. During the training for student you just need a BCE loss between the student's predictions and GTs, and a MSE loss between the refined feature maps of the teacher and student.
